### PR TITLE
Add primary and spouse dob to az name dob form #186973061

### PR DESCRIPTION
**Pull request by mrotondo**

### CODE REVIEW

**Comment by mrotondo** [comment]
last thing we talked about doing was removing StateFileAzIntake#ask_primary_dob? and its NY equivalent, since both return true now - and possibly move all these functions into StateFileBaseIntake since they're now identical in both states. The optional spouse dob test should also be removed from name_dob_form_spec

File: spec/lib/submission_builder/ty2022/states/return_header_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        it "generates xml with primary and spouse DOBs" do
+          doc = SubmissionBuilder::Ty2022::States::ReturnHeader.new(submission).document
+          expect(doc.at("Filer Primary DateOfBirth").text).to eq primary_birth_date.strftime("%F")
+          expect(doc.at("Filer Secondary DateOfBirth").text).to eq spouse_birth_date.strftime("%F")
+        end
+      end
+    end
+  end
+end
```
Comment (rickreyhsig) [inline spec/lib/submission_builder/ty2022/states/return_header_spec.rb:46]:
```suggestion
end

```

File: spec/forms/state_file/name_dob_form_spec.rb [hunk 2]
```diff
@@ -146,67 +154,16 @@ def params_with_values_from_intake(params, intake)
...
-          expect(second_dependent.dob).to eq Date.parse("January 11, 2013")
-          expect(second_dependent.months_in_home).to eq 10
-        end
+        expect(second_dependent.dob).to eq Date.parse("January 11, 2013")
       end
     end
 
     context "when married filing separately in new york" do
       let!(:intake) { create :state_file_ny_intake, filing_status: "married_filing_separately" }
+
```
Comment (rickreyhsig) [inline spec/forms/state_file/name_dob_form_spec.rb:166]:
```suggestion
```

**Review by rickreyhsig — APPROVED** [review]
lgtm

**Review by rickreyhsig — APPROVED** [review]
:shipit:

**Pull request merged**